actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-platform-core/actor-util/src/main/java/org/sunbird/actorutil/org/impl/OrganisationClientImpl.java

```diff
@@ -96,26 +105,33 @@ public Organisation getOrgById(ActorRef actorRef, String orgId, RequestContext c
     request.setRequest(requestMap);
     request.setOperation(ActorOperations.GET_ORG_DETAILS.getValue());
 
-    Object obj = interServiceCommunication.getResponse(actorRef, request);
-
-    if (obj instanceof Response) {
-      ObjectMapper objectMapper = new ObjectMapper();
-      Response response = (Response) obj;
-
-      // Convert contact details (received from ES) format from map to
-      // JSON string (as in Cassandra contact details are stored as text)
-      Map<String, Object> map = (Map) response.get(JsonKey.RESPONSE);
-      map.put(JsonKey.CONTACT_DETAILS, String.valueOf(map.get(JsonKey.CONTACT_DETAILS)));
-      organisation = objectMapper.convertValue(map, Organisation.class);
-    } else if (obj instanceof ProjectCommonException) {
-      throw (ProjectCommonException) obj;
-    } else if (obj instanceof Exception) {
-      throw new ProjectCommonException(
-          ResponseCode.SERVER_ERROR.getErrorCode(),
-          ResponseCode.SERVER_ERROR.getErrorMessage(),
-          ResponseCode.SERVER_ERROR.getResponseCode());
+    try {
+      Timeout t = new Timeout(Duration.create(10, TimeUnit.SECONDS));
+      Future<Object> future = Patterns.ask(actorRef, request, t);
+      Object obj = Await.result(future, t.duration());
+
+      if (obj instanceof Response) {
+        ObjectMapper objectMapper = new ObjectMapper();
+        Response response = (Response) obj;
+
+        // Convert contact details (received from ES) format from map to
+        // JSON string (as in Cassandra contact details are stored as text)
+        Map<String, Object> map = (Map) response.get(JsonKey.RESPONSE);
+        map.put(JsonKey.CONTACT_DETAILS, String.valueOf(map.get(JsonKey.CONTACT_DETAILS)));
+        organisation = objectMapper.convertValue(map, Organisation.class);
+      } else if (obj instanceof ProjectCommonException) {
```

Comment:
its better to just check for obj instanceof Response and move all remaining check to catch block 
like
catch (Exception e) {
 if (obj instanceof ProjectCommonException) {
      throw (ProjectCommonException) obj;
    } else if (obj instanceof Exception) {
      throw new ProjectCommonException(
          ResponseCode.SERVER_ERROR.getErrorCode(),
          ResponseCode.SERVER_ERROR.getErrorMessage(),
          ResponseCode.SERVER_ERROR.getResponseCode());
    }
}